docs/android/architecture.md

```diff
@@ -0,0 +1,98 @@
+
+### Build Logic
+
+The project uses multiple Gradle modules. Shared logic is centralized in a separate Gradle project named `build-logic`, included in the main project via `includeBuild`.
+
+### Common Gradle Module
+
+To share code across different applications, we use a common Gradle module named `:common`.
+
+## UI Development
+
+### Native UI
+
+All new UI components are built using [Jetpack Compose](https://developer.android.com/compose), ensuring a modern and declarative approach to UI development.
+
+### Legacy UI
+
+Some legacy XML layouts, `databinding`, and `viewbinding` still exist in the app. These should be replaced with Compose as part of ongoing modernization efforts.
+
+### Theming
+
+The app uses multiple themes to support both legacy XML and Compose-based UI. All new components should use `HomeAssistantAppTheme`, which is based on [Material Design](https://developer.android.com/develop/ui/compose/components).
+
+## Key Features
+
+### Dependency Injection (DI)
+
+We use [Hilt](https://developer.android.com/training/dependency-injection/hilt-android) extensively for dependency injection, ensuring modular and testable code.
+
+### Concurrency
+
+All concurrency is handled using [Kotlin Coroutines](https://kotlinlang.org/docs/coroutines-overview.html), providing a structured and efficient way to manage asynchronous tasks.
+
+### Services
+
+We use [Foreground Services](https://developer.android.com/develop/background-work/services/fgs) for retrieving sensor values and uploading them to Home Assistant Core asynchronously.
+
+### WebSocket
+
+The app maintains a direct connection to Home Assistant Core's WebSocket using [OkHttp](https://square.github.io/okhttp/). This is essential for features like Assist and real-time discussions.
+
+### REST API
+
+Communication with Home Assistant's REST API is handled using [Retrofit](https://square.github.io/retrofit/), enabling seamless interaction with the backend.
+
+### Local Storage
+
+- **Room**: User data is stored locally using [Room](https://developer.android.com/training/data-storage/room), which provides a robust database solution.
+- **SharedPreferences**: For app-specific settings, we use [SharedPreferences](https://developer.android.com/reference/android/content/SharedPreferences) with an abstraction layer named `LocalStorage`.
+
+### Deep Linking
+
+The app supports deep linking using `homeassistant://` URLs to navigate to specific parts of the app. For more details, refer to the [user documentation](https://companion.home-assistant.io/docs/integrations/url-handler/).
+
+## Platform-Specific Features
+
+### Automotive
+
+The automotive application reuses the sources of the `:app` module, simplifying development.
+
+### Wear OS
+
+The Wear OS app communicates with the mobile app via the [Messaging API](https://developer.android.com/training/wearables/data/messages). It only works with the `full` flavor, as it requires Google Play Services.
```

Comment:
It communicates with the mobile app, but only when required. It is standalone where possible.

Reply:
Yeah but I wrote it like this because I think you still need to full app to login the first time no? Especially because of the MessagingAPI. I might have missed it, when I set up mine I use the phone to login.

Reply:
Yes you need it for login (no webview on a watch, the frontend flow is the only supported one, and we can't do a device code flow because we don't know your url) and template settings. Optionally for managing some other settings.
The main features work without the companion app though, or theoretically without any phone connection, while to me this reads like _all_ communication happens via the Messaging API.

Reply:
Yea it is used for login as a requirement, its also used for the settings screens on the phone so users can do things like drag and drop to rearrange favorites on the home screen. A hard requirement is the template widget as that can ONLY be configured on the phone as the watch keyboard lacks keys used by templates `{` for example.

For things like sensor updates it does not use that method.

Reply:
I think the sentence is still valid. It doesn't communicate with the mobile except if it's using the `full` variant. Do you want me to rephrase it to make it clearer?

Reply:
Maybe make it clear the API is only used for setup and some configuration? We had some users request the watch to piggy back its data connection to the phone to transmit data but thats not how it works and this sentence may make it seem like the API is used for all communication.